src/ttf2adj.py

```diff
@@ -0,0 +1,179 @@
+                if (j == start or j == end) and FT_CURVE_TAG(
+                        tags[start]) != ON and FT_CURVE_TAG(tags[end]) != ON:
+                    on_count += 1
+                # allows end point of one curve to be start point of next curve
+                if on_count < 2:
+                    j += 1
+            if j > end:
+                break
+
+            # filling in Adjacency matrix
+            # all nodes in a curve are connected to each other
+            for ii in node_indices:
+                for jj in node_indices:
+                    if ii != jj:
+                        A[ii, jj, 2] = 1
+            # diagonal element =1 means on curve point
+            A[node_indices[0], node_indices[0], 2] = 1
+            A[node_indices[-1], node_indices[-1], 2] = 1
+
+            # nodes numpy array (2,num_pts)
+            nodes = points[:, node_indices]
+
+            if (j == start or j == end) and FT_CURVE_TAG(
+                    tags[start]) != ON and FT_CURVE_TAG(tags[end]) != ON:
+                midpoint = np.reshape((points[:, start] + points[:, end]) / 2,
+                                      (2, 1))
+                if j == start:
+                    nodes = np.insert(nodes, 0, midpoint, axis=1)
+                if j == end:
+                    nodes = np.concatenate((nodes, midpoint), axis=1)
+            if cubic_count == 1:
+                print(
+                    'too few cubic points for node_indices')
+            if conic_count > 1:
+                k = 1
+                mps = []
+                r = 1
+                while r < conic_count:
+                    r += 1
+                    midpoint = np.reshape((nodes[:, k] + nodes[:, k + 1]) / 2,
+                                          (2, 1))
+                    midpoint = np.repeat(midpoint, 2, axis=1)
+                    nodes = np.insert(
+                        nodes, k + 1, midpoint.transpose(), axis=1)
+                    mps.append(k + 2)
+                    k += 3
+
+                split_nodes = np.split(nodes, mps, axis=1)
+                curves.extend(split_nodes)
+
+            # appending complete bezier curve
+            else:
+                curves.append(nodes)
+
+        if FT_CURVE_TAG(tags[start]) == ON and FT_CURVE_TAG(tags[end]) == ON:
+            nodes = points[:, (start, end)]
+            curves.append(nodes)
+
+    # CONVERTING CURVE LIST TO 4xMx2 NUMPY ARRAY,
+    # WHERE M IS THE NUMBER OF CURVES IN THE GLYPH
+    np_glyph = np.empty((4, len(curves), 2), dtype=np.double)
+    for i in range(len(curves)):
+        # elevate all curvers to 3rd order (4 pts)
+        while curves[i].shape[1] < 4:
+            c = bezier.Curve.from_nodes(curves[i])
+            curves[i] = c.elevate().nodes
+        np_glyph[:, i, :] = curves[i].transpose()
+
+    return A
+
+
+
+def face2adj(fn, w, h):
+    face = Face(fn)
+    face.set_char_size(48 * 64)
```

Comment:
What are these numbers?

Reply:
Well based on what we saw today, 48 is the size of the adjacency matrix and 64 are the number of glyphs in the font. I think it is a good idea to make these as variables. Do you think the script should be doing this automatically or can we pass them as parameters?

Reply:
Can probably be done automatically. Take a look

Reply:
I get the impression that it's required to set this, but I have a feeling it doesn't do anything for the parts of the library that we're using. Would be nice to check if changing this value has any effect on the result